### PR TITLE
[SLA] PIM-5656 : Fix category tree does not load when 0 categories selected	

### PR DESCRIPTION
**Pull request by philippemossiere**
Fix category tree does not load when 0 categories selected

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | N
| Added Behats                      | Y
| Changelog updated                 | Y
| Review and 2 GTM                  | Y
| Micro Demo to the PO (Story only) |
| Migration script                  |
| Tech Doc                          |

### CODE REVIEW

File: src/Pim/Bundle/EnrichBundle/Resources/public/js/product/form/categories.js [hunk 1]
```diff
@@ -136,9 +143,16 @@ define(
              *
              * @param {String} rootTreeCode
              */
-            updateCategoryCount: function (rootTreeCode) {
+            updateCategoryCount: function (rootTreeCode, treeId) {
                 var $rootTreeContainer = this.$('li[data-code=' + rootTreeCode +  ']');
                 var selected = $rootTreeContainer.find('.jstree-checked');
+
+                if (selected.length === 0) {
```
Comment (solivier) [inline src/Pim/Bundle/EnrichBundle/Resources/public/js/product/form/categories.js]: yoda style

Comment (grena) [inline src/Pim/Bundle/EnrichBundle/Resources/public/js/product/form/categories.js]: I'd prefer to check for 
```
if (selected.length > 0) {
```

File: features/product/classify_product.feature [hunk 2]
```diff
@@ -49,3 +49,15 @@ Feature: Classify a product
...
+  Scenario: Change categories without saving
+    Given I edit the "tea" product
+    When I visit the "Categories" tab
+    And I select the "2014 collection" tree
+    And I expand the "2014 collection" category
+    And I click on the "Summer collection" category
+    And I click on the "Winter collection" category
+    Then I should see the text "There are unsaved changes."
+    When I visit the "Attributes" tab
+    Then I visit the "Categories" tab
```
Comment (jjanvier) [inline features/product/classify_product.feature]: and

**Comment by jjanvier** [comment]
gtm once fixed

File: features/product/classify_product.feature (hunk 2 above)
Comment (grena) [inline features/product/classify_product.feature]: And

File: features/product/classify_product.feature [hunk 3]
```diff
@@ -49,3 +49,15 @@ Feature: Classify a product
     And I click on the "123" category
     When I save the product
     Then I should see "Product successfully updated"
+
```
Comment (grena) [inline features/product/classify_product.feature]: Please add `@jira https://akeneo.atlassian.net/browse/PIM-5656`

File: CHANGELOG-1.4.md [hunk 4]
```diff
@@ -1,6 +1,7 @@
 # 1.4.x
 
 ## Bug fixes
+- PIM-5656: Category Tree does not load when 0 categories selected
```
Comment (grena) [inline CHANGELOG-1.4.md]: > 0 category

File: features/product/classify_product.feature (hunk 2 above)
Comment (willy-ahva) [inline features/product/classify_product.feature]: And

File: src/Pim/Bundle/EnrichBundle/Resources/public/templates/product/tab/categories.html [hunk 5]
```diff
@@ -3,7 +3,7 @@
         <ul class="nav nav-tabs" id="trees-list">
             <% _.each(trees, function (tree) { %>
                 <li data-tree="<%- tree.code %>" data-tree-id="<%- tree.id %>"<% if (tree.code === state.currentTree) { %> class="active"<% } %>>
-                    <a <% if (tree.associated) { %>class="tree-has-item"<% } %>href="javascript:void(0);" id="tree-link-<%- tree.id %>">
+                    <a href="javascript:void(0);" id="tree-link-<%- tree.id %>">
```
Comment (grena) [inline src/Pim/Bundle/EnrichBundle/Resources/public/templates/product/tab/categories.html]: no way to use your `treeLinkSelector`?

**Comment by grena** [comment]
Comments to fix then GTM

**Comment by willy-ahva** [comment]
GTM once fixed

File: src/Pim/Bundle/EnrichBundle/Resources/public/templates/product/tab/categories.html (hunk 5 above)
Comment (philippemossiere) [inline src/Pim/Bundle/EnrichBundle/Resources/public/templates/product/tab/categories.html]: Ok, I'll fix it

**Pull request merged**